app/validators/email_validator.rb

```diff
@@ -0,0 +1,9 @@
+# app/validators/email_validator.rb
+
+class EmailValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    unless value =~ /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i

```

Comment:
I don't suggest you use regex to validate email addresses.  see:

http://stackoverflow.com/questions/201323/using-a-regular-expression-to-validate-an-email-address

If you think about it, if it were possible to accurately validate email addresses, even very complicated, Rails would have had that built in the framework, because it is such a common task. Or at least there will be leading Ruby gems to do that. But there's really none that can cover all the edge cases. The solution should still be - just send an email for the user to confirm.